franklin/github/tests.py

```diff
@@ -1,62 +1,248 @@
-from unittest import mock
+from datetime import datetime
+import json
+
+from unittest.mock import Mock, patch
 
 from django.contrib.auth.models import User
-from django.test import TestCase, RequestFactory
+from django.core.urlresolvers import reverse
 
-from rest_framework import status
+from rest_framework.test import APITestCase
 
-from .views import get_auth_token
-from builder.models import Owner, Site
+from builder.models import BranchBuild, Build, Deploy, Environment, Owner, Site
 
 
-class ConfigTestCase(TestCase):
-    def setUp(self):
-        self.user = User.objects.create(username="test", password="asdf")
-        social = self.user.social_auth.create(provider='github')
-        social.extra_data['access_token'] = ''
-        social.save()
-        self.owner = Owner.objects.create(
-            name='istrategylabs', github_id=607333)
-        self.site = Site.objects.create(
-            owner=self.owner, name='franklin-dashboard', github_id=45864453)
+def ordered(obj):
+    if isinstance(obj, dict):
+        return sorted((k, ordered(v)) for k, v in obj.items())
+    if isinstance(obj, list):
+        return sorted(ordered(x) for x in obj)
+    else:
+        return obj
```

Comment:
I didn't want to do this, but for some reason the dict returned from the api response is slightly different, so you can't compare them directly. You have to order the objects first. (see 'case in point' below)